gflownet/gflownet.py

```diff
@@ -236,20 +236,19 @@ def forward_sample(
             .to(bool)
         )
         # check for at least one non-random action
-        if idx_norandom.sum() > 0:
-            if sampling_method == "policy":
-                policy_outputs[idx_norandom, :] = model(
-                    self._tfloat(
-                        self.env.statebatch2policy(
-                            [s for s, do in zip(states, idx_norandom) if do]
-                        )
+        if sampling_method == "policy" and idx_norandom.sum() > 0:
```

Comment:
ah I thought we might want to nest both conditions `policy` and `uniform` under the `idx_norrandom` check. Or we can add a `#TODO` and be mindful while writing the logic for uniform sampling